### PR TITLE
Added ability to send reports using PUT

### PR DESCRIPTION
**Pull request by victor-chaparro**
This PR adds the ability to send reports using PUT through the REST transport and also sets the accessToken if provided in the credential.
Minnesota's REST API only accepts reports be sent to them using PUT. The `transport.headers.method` setting must be set to `PUT` so that the REST transport can configure the HTTP client to send via PUT, otherwise it will use POST by default.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Run `./prime multiple-settings set --input ./settings/STLTs/MN/mn-doh.yml`
2. Send test report routed to MN
3. Verify the report gets sent to MN
4. Query the `action` table by running `select * from action order by created_at desc`. Wait for an `action_name` with `send_error` entry to show up. The report fails because we're sending to a mock API that only validates the message and our messages fail that validation. The errors will be fixed, but for now this is only to test we can send reports to their REST API. We can verify that by looking at the `action_result` and making sure we get validation errors back.

## Changes
- Added ability to specify REST method inside transport.headers

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

## Linked Issues
- Fixes #14696

### CODE REVIEW

**Review by JessicaWNava — APPROVED** [review]
Couple of minor changes. Mostly I need to know how to verify that the message sent to them. I didn't see any errors when I sent it, but is there another way to verify?

**Review by JessicaWNava — CHANGES_REQUESTED** [review]
Oops meant to request changes

**Review by JessicaWNava — APPROVED** [review]
Changes good, just waiting on instructions to figure out how to verify that the message actually sent.

**Comment by JessicaWNava** [comment]
When I tested this I did not see a send_error action in the table. Seems to be stuck on waiting to deliver despite the numberPerDay being set to 1440

File: prime-router/src/main/kotlin/transport/RESTTransport.kt [hunk 1]
```diff
@@ -592,84 +595,106 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
...
-                        if (key == "Content-Length" && value == "<calculated when request is sent>") {
-                            Pair(key, message.size.toString())
-                        } else {
-                            Pair(key, value)
-                        }
-                    }.toMap()
-                )
+            val response: HttpResponse = if (headers["method"] == "PUT") {
+                client.put(restUrl) {
+                    build(logger, headers, message, fileName, boundary, reportCreateDate)
```
Comment (GilmoreA6) [inline prime-router/src/main/kotlin/transport/RESTTransport.kt:601]: can we add a test to cover this logic split here?

**Review by GilmoreA6 — APPROVED** [review]

**Pull request merged**